### ISSUE TITLE
Edit-build-run loop when modifying Python is a bit cumbersome

### ISSUE THREAD

**Issue by bbqsrc** [indygreg/PyOxidizer#296]
Every time I modify Python code, if I want it to be included in the build while testing, I have to delete `build` and do `cargo clean` or the new modifications aren't picked up.

Is there a cleaner approach?

**Comment by indygreg** [comment]
I agree that the edit-build-run loop here is a bit lacking. I have a few ideas to address this:

* Better dependency out-of-date detection (this existed once before but regressed a release or 2 ago)
* Better support for standalone resources files. The long pole in building is producing an executable. If we supported standalone resources files, we wouldn't have to link a new binary every time a Python resource changed.

You could work around this today with a few dozen lines of custom code using the `oxidized_importer` Python API. See https://pyoxidizer.readthedocs.io/en/stable/oxidized_importer_freezing_applications.html for inspiration. You could build an executable that runs a file which invokes `oxidized_importer` to scan for new resources and make them available to the interpreter. Startup time would be slow due to the dynamic resource scanning. But it would likely be faster than building/linking a new binary!
